### PR TITLE
Flexible con grouping.

### PR DESCRIPTION
**Pull request by pravsripad**
Some changes to start with #260 . 

1. Do we really need an ordered dict? 
2. Exposed using cortex colors from different colormaps as well.

### CODE REVIEW

**Comment by ckiefer0** [comment]
We need to keep the same order of the keys because otherwise the order of the groups might randomly change. I think that's why there is this weird implementation of lists of dicts. However, in the [newer python versions](
https://stackoverflow.com/questions/1867861/how-to-keep-keys-values-in-same-order-as-declared), dicts apparently do keep the order of the keys. The question would then be backwards compatibility.

File: jumeg/connectivity/con_viz.py [hunk 1]
```diff
@@ -561,22 +561,34 @@ def plot_grouped_connectivity_circle(yaml_fname, con, orig_labels,
...
+        # read the yaml file with grouping
+        if op.isfile(yaml_fname):
+            with open(yaml_fname, 'r') as f:
+                label_groups = yaml.safe_load(f)
+        else:
+            print('%s - File not found.' % yaml_fname)
+            sys.exit()
+    elif isinstance(yaml_fname, list):
+        # should be a list of dictionaries
+        label_groups = yaml_fname
```
Comment (ckiefer0) [inline jumeg/connectivity/con_viz.py:574]: I find it a bit unintuitive that yaml_fname could be a list. I guess that this is in preparation for getting rid of yamls but then we should rename the variable.

Also, now that I think about it, we should just make this a small function for reading the groups from the yaml instead of repeating the code twice in a row.

**Comment by pravsripad** [comment]
> We need to keep the same order of the keys because otherwise the order of the groups might randomly change. I think that's why there is this weird implementation of lists of dicts. However, in the [newer python versions](https://stackoverflow.com/questions/1867861/how-to-keep-keys-values-in-same-order-as-declared), dicts apparently do keep the order of the keys. The question would then be backwards compatibility.

As I understand it, when we use a list of dict, the order of the groups are preserved because it is a list. I think the "list of dict" was implemented instead of dict because previously dict did not preserve order. Now we have the option to change "list of dict" to dict/OrderedDict but there is no real reason to do so.

**Comment by pravsripad** [comment]
@ckiefer0 can you please have a look again? If you are okay with the changes then we can merge and in a new PR modify examples to reflect variable name changes if needed.

**Comment by ckiefer0** [comment]
> @ckiefer0 can you please have a look again? If you are okay with the changes then we can merge and in a new PR modify examples to reflect variable name changes if needed.

In general, the changes look good. However, due to the different order of functions it is extremely hard to follow any changes in con_viz.py. In the future it might be a better idea to leave that to a separate pull request where there are no changes to the code itself.

**Comment by pravsripad** [comment]
> > @ckiefer0 can you please have a look again? If you are okay with the changes then we can merge and in a new PR modify examples to reflect variable name changes if needed.
> 
> In general, the changes look good. However, due to the different order of functions it is extremely hard to follow any changes in con_viz.py. In the future it might be a better idea to leave that to a separate pull request where there are no changes to the code itself.

Yeah I should have kept that for a last commit, sorry about that.

Thanks a lot @ckiefer0 . Merging it now.
